scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -149,252 +198,377 @@ sce_list
+In our case, we want to run `format_sce()` over paired `sce_list` items and `sce_list` names.
 
 ```{r format sces for merge}
-sce_list_formatted <- purrr::imap(
+# We can use `purrr::map2()` to loop over two list/vector arguments simultaneously
+sce_list_formatted <- purrr::map2(
+  # Each "iteration" will march down the first two 
+  #  arguments `sce_list` and `names(sce_list)` in order
   sce_list,
-  format_sce,
-  shared_genes = shared_genes
+  names(sce_list),
+  # Name of the function to run
+  format_sce
 )
+
 sce_list_formatted
 ```
+(Psst, like `purrr` and want to dive deeper? Check out [the `purrr::imap()` function](https://purrr.tidyverse.org/reference/imap.html)!)
+
 
-We are now ready to merge the SCEs!
-We'll use the R function `cbind()` to combine the SCE objects.
-The `cbind()` function is often used to combine data frames by column, i.e. "stack" them on top of each other.
-The same principle applies here, but when run on SCE objects, `cbind()` will create a new SCE object by combining `counts` and `logcounts` matrices by column, as well as combining reduced dimension slots appropriately.
+At long last, we are ready to merge the SCEs!
+We'll use the R function `cbind()` for this.
+The `cbind()` function is often used to combine data frames or matrices by column, i.e. "stack" them next to of each other.
+The same principle applies here, but when run on SCE objects, `cbind()` will create a new SCE object by combining `counts` and `logcounts` matrices by column, as well as combining metadata and reduced dimension slots appropriately.
 
 Since we need to apply `cbind()` to a _list_ of objects, we need to use some slightly-gnarly syntax: We'll use the function `do.call()`, which allows the `cbind()` input to be a list of objects to combine.
 
 ```{r merge sces}
+# Merge SCE objects 
 merged_sce <- do.call(cbind, sce_list_formatted)
+
+# Let's have a look!
 merged_sce
 ```
 We now have a single SCE object that contains all cells from all libraries we'd like to integrate.
 
+Let's take a peek at some of the innards of this new SCE object:
+```{r explore merged_sce}
+# What are the unique values in the `library` column?
+unique( colData(merged_sce)$library )
+
+# What are the new cell ids (column names)?
+head( colnames(merged_sce) )
+
+# What does rowData look like?
+head( rowData(merged_sce) )
+```
+
+TODO: Something else that might be fun is to have patient information and toss onto UMAPs, but with only 4 libraries already ordered by patient pairs, this is easy enough to do by eye in a batch-level UMAP so I don't think we necessarily need this. It could be useful to show though?
+
+```{r add patient metadata}
+merged_sce$patient <- dplyr::case_when(
+  merged_sce$library %in% c("SCPCL000479", "SCPCL000480") ~ "A",
+  merged_sce$library %in% c("SCPCL000481", "SCPCL000482") ~ "B"
+  
+)
```

Comment:
Do you use this at all below? I think right now I would remove it from this pre-integration chunk and if we do want to make any plots with it then it would be something you could modify right before plotting. Also another thought is that this is something we add into the exercises...

Reply:
No, this chunk isn't currently used, just an idea! I will remove it. It may or may not end up being relevant for exercises depending on what samples get used (CC @jashapiro who is writing integration exercises).

Reply:
I wonder if this is worth including because it does show how something you might include in a harmony model. I'm not sure it is worth the effort, unless it improves the harmony integration.